samples/secure-fraud-detection/src/main/java/io/quarkiverse/langchain4j/sample/ChatMemoryBean.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkiverse.langchain4j.sample;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import jakarta.annotation.PreDestroy;
+import jakarta.enterprise.context.RequestScoped;
+
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.ChatMemoryProvider;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+
+@RequestScoped
+public class ChatMemoryBean implements ChatMemoryProvider {
```

Comment:
Isn't this implemented exactly the same as the default memory provider? Why is this needed then?

Reply:
@jmartisk It was only a copy and paste, I think it was added in the original fraud-detection demo just to should how it can be customized. I'm not sure yet it will be needed for the secure variation of the demo

Reply:
@jmartisk @geoand I've dropped this custom chat memory provider from the demo as a new memory id is provided on every request, so indeed it is not clear to me how it can be used in this specific demo, but a custom memoryId provider will definitely be of interest here as I'm having an authenticated session, so I can use exactly the same memory id to correlate several fraud amount and distance requests.
So I'll try to add it here with another commit and ping you